src/mdlComponentHandler.js

```diff
@@ -86,6 +86,15 @@ var componentHandler = (function() {
    * the element to.
    */
   function upgradeElementInternal(element, jsClass) {
+    if (jsClass === undefined) {
+      for (var i = 0; i < registeredComponents_.length; i++) {
```

Comment:
FYI I'm changing the other foreach in this method to for in #895 
Please review #895 ;)